dnf/base.py

```diff
@@ -1556,7 +1556,10 @@ def upgrade_all(self, reponame=None):
             self._goal.upgrade_all()
         else:
             try:
-                self.upgrade('*', reponame)
+                rent=self.sack.query().installed()
+                for item in rent:
+                    if "@"+reponame == item.from_repo:
```

Comment:
this is not IMO right. You don;t want to upgrade packages that were downloaded from exact repo. You want to upgrade all available in repo.